LaunchDarkly/LaunchDarkly/LDClient.swift

```diff
@@ -827,24 +827,25 @@ public class LDClient {
     }
 
     static func start(serviceFactory: ClientServiceCreating?, config: LDConfig, context: LDContext? = nil, startWaitSeconds: TimeInterval, completion: ((_ timedOut: Bool) -> Void)? = nil) {
-        var completed = true
+        var completed = false
         let internalCompletedQueue: DispatchQueue = DispatchQueue(label: "TimeOutQueue")
         if !config.startOnline {
             start(serviceFactory: serviceFactory, config: config, context: context)
-            completion?(completed)
+            completion?(true) // offline is considered a short circuited timed out case
         } else {
             let startTime = Date().timeIntervalSince1970
             start(serviceFactory: serviceFactory, config: config, context: context) {
                 internalCompletedQueue.async {
-                    if startTime + startWaitSeconds > Date().timeIntervalSince1970 && completed {
-                        completed = false
-                        completion?(completed)
+                    if startTime + startWaitSeconds > Date().timeIntervalSince1970 && !completed {
```

Comment:
I think you could just remove the `startTime + startWaitSeconds > Date().timeIntervalSince1970` condition

Reply:
And if this closure runs before the one passed to `asyncAfter` but `startTime + startWaitSeconds > Date().timeIntervalSince1970` is false (e.g. because of the leeway that dispatch allows with `asyncAfter`) it seems unnecessary to fail just due to time if you know you've actually succeeded.